src/event/workqueue.c

```diff
@@ -218,28 +218,27 @@ _dispatch_workq_monitor_pools(void *context DISPATCH_UNUSED)
 			_dispatch_debug("workq: %s under utilization target; poking with floor %d",
 					dq->dq_label, floor);
 			_dispatch_global_queue_poke(dq, 1, floor);
+			global_runnable += 1; // account for poke in global estimate
 		}
 	}
 }
 #endif // HAVE_DISPATCH_WORKQ_MONITORING
 
-
-// temporary until we switch over to QoS based interface.
 static dispatch_queue_t
-get_root_queue_from_legacy_priority(int priority)
+get_root_queue_from_dispatch_qos(dispatch_qos_t qos)
```

Comment:
this seems effectively the same as _dispatch_get_root_queue now ? (inline_internal.h)
can we not use that here ?